### ISSUE TITLE
redundant coma between last argument and `do`

### ISSUE THREAD

**Issue by sleewoo** [somebee/imba#36]
Imho, `save data do`  looks much more natural than `save data, do`

i understand it's just a caprice but permanently switching between Ruby and Imba makes me forget about that coma.

and when adding it my eyes ache :)

**Comment by somebee** [comment]
I agree. Imba has different precedence than Ruby when it comes to blocks. I've reasoned about it a few times, and have come to the conclusion that I like the Imba precedence better, but I cannot really remember why. Looks like we should go over it again.

By the way, how it works in imba:
```imba
# It will only happen with callable arguments
save data do yes # block is called on data
save "data" do yes # block is called on save
save "data", 100 do yes # still called on save

# I personally never use comma, but use parens around the non-block args:
save(data) do yes
```
I need to go over this again and try to remember why this was supposedly a better idea :) I did discuss this briefly with @judofyr, maybe he can remember the reasoning?

**Comment by sleewoo** [comment]
in most cases i use parens too but for people who prefer parens-less syntax it's a useless payload to recall to put coma if last arg is callable.

**Comment by somebee** [comment]
The reason I'm leaning towards the existing syntax (besides compatibility) is the fact that it allows for this type of pattern, which I find myself using from time to time:
```imba
load something.map do |item|
    item.path
```
If we had ruby-like precendence, the block *must* be wrapped:
```imba
load something.map(do |item|
    item.path
)
```
I find it a bit clunkier to need to wrap the block itself, than to sometimes need to wrap the preceding arguments (or use a single comma). It seems easier to control where the do block is with the current syntax, but it's what I am used to, so hard to say.

**Comment by judofyr** [comment]
One "solution" would be to disallow ambiguous `do` blocks:

```
# ok:
load(something.map) do |item|
    item.path
# ok:
load(something.map do |item|
    item.path)
# ok:
load something.map(do |item|
    item.path)
# syntax error:
load something.map do |item|
    item.path
```

**Comment by sleewoo** [comment]
+1 on disallowing ambiguous `do` blocks
